### ISSUE TITLE
Several accessibilities roles are not working | Android

### ISSUE THREAD

**Issue by nico22utn** [facebook/react-native#30807]
Please provide all the information requested. Issues that do not follow this format are likely to stall.

## Description
Currently, I've implemented the accessibilities feature for a RN app (0.63.4)
And I've noticed that several accessibilities roles are not working anymore.
It is happening only for Android.
I 've just tested for Samsung s10e and I was able to reproduce it.
Testing with samsung j6 is working as expected.

This PR could be related: https://github.com/facebook/react-native/pull/27496/files
## React Native version:
0.63.4

## Steps To Reproduce
Just Enable Talkback. You can follow this link(https://reactnative.dev/docs/accessibility) in order to enable the Talkback for the emulator
Read some element with the accessibiltyRole='header' or 'text' or 'keyboardkey'

## Expected Results
The app should be able to read the different accessibilities roles.

**Comment by safaiyeh** [comment]
Thank you @nico22utn for the issue & the investigation. Could you provide a repro and exact steps to test it with TalkBack so it is easier to investigate the issue. Once I can confirm it locally I'll tag the appropriate people.

PR ref: https://github.com/facebook/react-native/pull/27496

**Comment by nico22utn** [comment]
> Thank you @nico22utn for the issue & the investigation. Could you provide a repro and exact steps to test it with TalkBack so it is easier to investigate the issue. Once I can confirm it locally I'll tag the appropriate people.
> 
> PR ref: #27496

That happens in the Android devices. Just set in some view or text the following:
`accessibilityRole="header"` and that's it.
I think that It is not necessary a repo. Just tell me if you need more information.

**Comment by amarlette** [comment]
This issue will be resolved by #30839 as part of the Improved React Native Accessibility Project

**Comment by chrisglein** [comment]
Duplicate of #30839
